mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -158,6 +158,9 @@ class Word2Vec extends Serializable with Logging {
       .sortWith((a, b) => a.cn > b.cn)
     
     vocabSize = vocab.length
+    require(vocabSize > 0, "The vocabulary size should be large than 0. You may need to check " +
```

Comment:
Yeah, this is real minor point of taste. I use `IllegalStateException` when there's no real argument to speak of, which is marginally more specific and description. But isn't this error caused by a bad input RDD? `IllegalArgumentException` seems like just the thing.